agent/local/state.go

```diff
@@ -325,10 +325,27 @@ func (l *State) RemoveServiceWithChecks(serviceID structs.ServiceID, checkIDs []
 	return nil
 }
 
+// Checks to see whether the given service ID matches the name of an existing
+// service. Used to see whether a service lookup failed because the user
+// mistakenly provided a service name rather than a service ID.
+func (l *State) doesServiceExistWithNameOfID(lookupID structs.ServiceID) bool {
```

Comment:
(still learning, feel free to disregard)

my .02 cents here is that **we could** do such a lookup but the philosophical question is whether we **we should** do it. I'm leaning towards "we shouldn't do this" as it introduces complexity/ surprises on our API path.

I think the right way here is to improve the err msg like Matt suggested:

https://github.com/hashicorp/consul/pull/10894/files#r704624702

Reply:
I think everyone is agreed on intent: that we should give the user an indication of what they might have done wrong.

The difference of opinion is purely on embodiment. It's certainly simpler to give one error message in all cases. It also avoids the need to consider [Mark's good questions](https://github.com/hashicorp/consul/pull/10894/files#r704630005). I think I've been convinced to go with the "1 common error message" approach, and may need to extend that to 1-2 other places based on the other points raised (e.g., interaction with ACLs).

Thanks everyone!